django_airavata/apps/workspace/static/django_airavata_workspace/js/components/storage/storage-edit/UserStorageLink.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <b-link :href="storageFileViewRouteUrl()" @click="showFilePreview($event)">
+    {{ fileName }}
+    <b-modal :title="fileName" ref="modal" scrollable size="lg">
+      <user-storage-file-edit-viewer
+        :file-name="fileName"
+        :data-product-uri="dataProductUri"
+        :mime-type="mimeType"
+        @file-content-changed="
+          (fileContent) => $emit('file-content-changed', fileContent)
+        "
+      />
+      <template slot="modal-footer">
+        <a :href="storageFileViewRouteUrl()" target="_blank">Open in a new window</a>
+      </template>
+    </b-modal>
+  </b-link>
+</template>
+
+<script>
+import UserStorageFileEditViewer from "./UserStorageEditViewer";
+
+export default {
+  name: "user-storage-link",
+  components: {UserStorageFileEditViewer},
+  props: {
+    fileName: {
+      required: true
+    },
+    dataProductUri: {
+      required: true
+    },
+    mimeType: {
+      required: true
+    },
+    allowPreview: {
+      default: true,
+      required: true
+    }
+  },
+  methods: {
+    showFilePreview(event) {
+      if (this.allowPreview) {
+        this.$refs.modal.show();
+        event.preventDefault();
+      }
+    },
+    storageFileViewRouteUrl() {
+      // This endpoint can handle XHR upload or a TUS uploadURL
+      return `/workspace/storage/~?dataProductUri=${this.dataProductUri}`;
```

Comment:
It would be nice if we could keep the regular URL path and pass the dataProductUri as a prop instead.

Reply:
@machristie dataProductUri  is a prop already passed to this component

https://github.com/apache/airavata-django-portal/pull/52/files/3c049283cb5211f76c4576251ffb59e55afd7a52#diff-589038988d47ba2be64cd940ebaa158b28630aa899bf0104ab66d9c69b09b2a9R30

Reply:
No I'm talking about the link produced here. The link should be the path to the file, like before https://github.com/apache/airavata-django-portal/blob/develop/django_airavata/apps/workspace/static/django_airavata_workspace/js/components/storage/UserStoragePathViewer.vue#L209.

Well, I guess, I don't quite understand why the dataProductUri needs to be a query parameter. Looks like UserStorageContainer uses it to get the path, but it wouldn't need that if the path were used like before. Can you explain why dataProductUri is needed as a query parameter?

Reply:
@machristie Passing dataProductUri as a query parameter is only an additional thing and passing the path is still working as well. I did so because I do not have the path in the URL when the file editor is opened through a dialog.